src/nncf/quantization/algorithms/weight_compression/algorithm.py

```diff
@@ -983,6 +1001,24 @@ def apply(
             },
             algo_name="weight_compression",
         )
+
+        return transformed_model
+
+    def apply(
+        self,
+        model: TModel,
+        graph: NNCFGraph,
+        statistic_points: Optional[StatisticPointsContainer] = None,
+        dataset: Optional[Dataset] = None,
+    ) -> TModel:
+        self.set_backend_entity(model)
+        nodes_to_compress = self.get_nodes_to_compress(graph)
+        # Get processed weight compression parameters ready for compression
+        all_weight_params, statistics = self.get_weight_compression_parameters(
+            model, graph, nodes_to_compress, statistic_points, dataset
+        )
+        transformed_model = self.apply_wc_algos(model, graph, all_weight_params, statistics, dataset)
```

Comment:
Please come up with a docstring as well
```suggestion
        transformed_model = self.apply_with_compression_parameters(model, graph, all_weight_params, statistics, dataset)
```